pallets/communities/src/lib.rs

```diff
@@ -386,5 +421,46 @@ pub mod pallet {
 			// Return a successful DispatchResultWithPostInfo
 			Ok(())
 		}
+
+		/// Enroll an account as a community member. In theory,
+		/// any community member should be able to add a member. However, this
+		/// can be changed to ensure it is a privileged function.
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::add_member())]
+		pub fn add_member(
+			origin: OriginFor<T>,
+			community_id: T::CommunityId,
+			who: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			Self::ensure_active(&community_id)?;
+			Self::ensure_member(origin, &community_id)?;
+
+			let who = <<T as frame_system::Config>::Lookup as StaticLookup>::lookup(who)?;
+			Self::do_insert_member(&community_id, &who)?;
+
+			Ok(())
+		}
+
+		/// Removes an account as a community member. While
+		/// enrolling a member into the community can be an action taken by any
+		/// member, the decision to remove a member should not be taken
+		/// arbitrarily by any community member. Also, it shouldn't be possible
+		/// to arbitrarily remove the community admin, as some privileged calls
+		/// would be impossible to execute thereafter.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::remove_member())]
+		pub fn remove_member(
+			origin: OriginFor<T>,
+			community_id: T::CommunityId,
+			who: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			Self::ensure_active(&community_id)?;
+			Self::ensure_privileged(origin, &community_id)?;
```

Comment:
This one should go first, to avoid the DB read if we already know the origin is not signed(or maybe we don't even need to check if the community is active for removing members? 🤔).  

Most times the check of checking the origin is the first thing you do, something I don't love about moving those checks to some inner functions is that is not evident first hand that the check of the origin is being made

Reply:
I see. A possible improvement here is to make more explicit where inner functions imply ensuring some origin. So, here's my proposal:

1. Naturally, moving the `ensure_privileged to the top of the calls` to avoid DB calls (although these origins internally make some DB reads if the origin is a proper `RuntimeOrigin::Signed`).
2. Renaming the functions to be more explicit in mentioning that they make ensuring of origins, like this:

```diff
-			Self::ensure_privileged(origin, &community_id)?;
+			Self::ensure_origin_privileged(origin, &community_id)?;
```

```diff
-			Self::ensure_member(origin, &community_id)?;
+			Self::ensure_origin_member(origin, &community_id)?;
```

Now, when discussing about not checking whether the community is active for removing members, I was thinking about some cases where the community should be frozen (by decision of the same community) or locked (by decision of the network) —maybe those names might be reconsidered as whether defining what's a frozen/locked community— but in those cases there might be a room to disallow the community state to change (maybe a frozen community means there's been an external intervention that requires freezing assets and retaining the state of it, maybe a regulator authority requiring that?).